optuna/trial.py

```diff
@@ -237,15 +241,23 @@ def objective(trial):
                 If ``log`` is true, the value is sampled from the range in the log domain.
                 Otherwise, the value is sampled from the range in the linear domain.
                 See also :func:`suggest_uniform` and :func:`suggest_loguniform`.
+            step:
+                A step of discretization.
 
         Returns:
             A suggested float value.
         """
 
-        if log:
-            return self.suggest_loguniform(name, low, high)
+        if step is not None:
+            if log:
+                raise NotImplementedError("The parameter step is not supported when log=True.")
```

Comment:
How about adding quotes to the argument names as can be seen in [distribution.py](https://github.com/optuna/optuna/blob/master/optuna/distributions.py#L313)?
 
```suggestion
                raise NotImplementedError("The parameter `step` is not supported when `log` is True.")
```

Reply:
Thank you for catching, I revised my PR.